iml-api/src/graphql.rs

```diff
@@ -3,26 +3,60 @@
 // license that can be found in the LICENSE file.
 
 use crate::{command::get_command, error::ImlApiError};
-use chrono::{DateTime, SecondsFormat, Utc};
+use chrono::{DateTime, Utc};
 use futures::{TryFutureExt, TryStreamExt};
 use iml_postgres::{sqlx, sqlx::postgres::types::PgInterval, PgPool};
 use iml_rabbit::Pool;
 use iml_wire_types::{snapshot::Snapshot, Command};
 use itertools::Itertools;
 use juniper::{
     http::{graphiql::graphiql_source, GraphQLRequest},
-    EmptySubscription, FieldError, GraphQLEnum, ParseScalarResult, ParseScalarValue, RootNode,
-    Value,
+    EmptySubscription, FieldError, GraphQLEnum, RootNode, Value,
 };
-use std::ops::Deref;
 use std::{
     collections::{HashMap, HashSet},
     convert::{Infallible, TryFrom as _},
+    ops::Deref,
     sync::Arc,
     time::Duration,
 };
 use warp::Filter;
 
+struct GraphQLDuration(Duration);
+
+#[juniper::graphql_scalar(
+    name = "Duration",
+    description = "Duration in human-readable form, like '15min 2ms'"
+)]
+impl<S> GraphQLScalar for GraphQLDuration
+where
+    S: juniper::ScalarValue,
+{
+    fn resolve(&self) -> juniper::Value {
+        juniper::Value::scalar(humantime::format_duration(self.0).to_string())
+    }
+
+    fn from_input_value(value: &juniper::InputValue) -> Option<GraphQLDuration> {
+        let x = value
```

Comment:
Just wondering if we could do something like:
```
value
    .as_string_value()?
.parse::<humantime::Duration>()
.ok()
.map(|x| GraphQLDuration(x))
```
?

Reply:
yeah, it could be all combinators.